ext/storage_sources/gcp_store/gcp_connection.cpp

```diff
@@ -85,15 +90,46 @@ gcp_connection::get_object(const std::string &object_key, const std::string &pat
 // Checks whether an object with the given key exists in the google cloud bucket and also retrieves
 // size of the object.
 int
-gcp_connection::object_exists(
-  const std::string &object_key, bool &exists, size_t &object_size) const
+gcp_connection::object_exists(const std::string &object_key, bool &exists, size_t &object_size)
 {
+    object_size = 0;
+
+    google::cloud::StatusOr<gcs::ObjectMetadata> mt =
+      _gcp_client.GetObjectMetadata(_bucket_name, object_key);
+    google::cloud::StatusCode code = mt.status().code();
```

Comment:
Have you tested this code? `mt` doesn't look to be a pointer but you are making a null check a few lines down, despite accessing its member functions here.

Reply:
Please rethink the logic for this function. Let me know if you have questions!

Reply:
Yep, `!mt` checks whether retrieving the bucket metadata produce an error. It will not be empty even when the bucket doesn't exist. For instance, if a bucket exists but we do not have permission to access it, the bucket metadata status will print out the error message `PERMISSION_DENIED: Permanent error GetBucketMetadata:` yet it still passes the condition `!mt`.  Since we only want to check if the bucket exists in this function, any other errors detected will be checked by the `!mt` condition here and return -1.

Reply:
I've read the `StatusOr` documentation. I suggest using `!mt.ok()` rather than `!mt` to avoid confusion and it would be useful to add a description of what the `StatusOr` function does. My understanding is that a `StatusOr` holds either a value or a status indicating why there is no value. The status is always accessible, but you need to check that `mt.ok()` is true before accessing the value. 

If a bucket does not exist and you do not have permission to access it, will the status still be `NOT_FOUND`?

Reply:
In the [newest version](https://github.com/wiredtiger/wiredtiger/pull/8737/commits/e806560cfcd70355152938a098de63b3c5071635), bucket exist is checked inside the constructor and it will throw an error if the bucket does not exist or is not accessible (i.e. `!metadata`). A boolean value is not needed in this case so we also won't need to use `metadata.ok()`. This would make the code much cleaner than before. 😄